### PR TITLE
shader: add sb_mode 0 in smp

### PR DESCRIPTION
**Pull request by Zangetsu38**
add case 0 request by game, fix regression  #1050

### CODE REVIEW

File: vita3k/shader/src/translator/texture.cpp [hunk 1]
```diff
@@ -181,11 +181,10 @@ bool USSETranslatorVisitor::smp(
     spv::Id result = do_fetch_texture(sampler, { coord, static_cast<int>(DataType::F32) }, DataType::F32);
 
     switch (sb_mode) {
-    case 1: {
+    case 0:
```
Comment (VelocityRa) [inline vita3k/shader/src/translator/texture.cpp:184]:
```suggestion
    case 0: // TODO: is this correct?
```

Comment (Zangetsu38) [reply vita3k/shader/src/translator/texture.cpp:184]: Not sur i wating @sunho info, but is user like that before is probably same mode 1

Comment (VelocityRa) [reply vita3k/shader/src/translator/texture.cpp:184]: ok let's see what sunho says

**Comment by pent0** [comment]
Sb_mode is sample bypassing mode. It's to decide which sample values to gather during the smp call. [Docs](http://cdn.imgtec.com/sdk-documentation/PowerVR+Instruction+Set+Reference.pdf) here talks a bit about it.

From @sunho work:
- mode 3 has some extra info (4 components), and mode 1 is just normal sample result.
- Based on the docs there, I assume mode 0 is default mode, same as mode 1.
- Mode 2 is info mode (contains sample info, i dont know what it contains yet, need hardware test. I guess it has mipmap level maybe).
- Mode 3 is both, since 4 latter components are sample data, 4 beginning components should be info mode

Add mode 0 fix rainy texture in menu of Virtua Tennis 4. All results gather here are likely correct.

a

lol i see zang already made pr

File: vita3k/shader/src/translator/texture.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/shader/src/translator/texture.cpp:184]: Is approuve

**Review by frangarcj — APPROVED** [review]

**Comment by sunho** [comment]
:+1::+1::+1:

**Pull request merged**